misc/h2olog/quic.h

```diff
@@ -0,0 +1,55 @@
+/*
```

Comment:
Can we just use `deps/quicly/include/quicly.h` instead?

Reply:
Right. It looks better because `gen-quic-bpf.py` already has some allow-lists and deny-lists to control which fields are handled or not. This `quic.h` declares which fields `h2olog` uses, but this kind of metadata should be moved to `gen-quic-bpf.py`.

Reply:
Talked about it with @kazuho and I don't tackle it in this PR, since it needs `quicly.h` in *runtime*. When h2o installs quicly and picotls headers in the future, then h2olog will use them.